src/nextjournal/clerk/builder.clj

```diff
@@ -183,8 +187,30 @@
       (expand-paths {:paths-fn `my-paths}))
 #_(expand-paths {:paths ["notebooks/viewers**"]})
 
+(defn compile-css!
+  "Compiles a minimal tailwind css stylesheet with only the used styles included, replaces the generated stylesheet link in html pages."
+  [{:keys [out-path]} docs]
+  (assert (and (= 0 (:exit (sh "which" "npx"))) (= 0 (:exit (sh "npx" "tailwindcss"))))
+          "Clerk's CSS optimizaiton failed: node and tailwind need to be installed. Please run `npm install -D tailwindcss @tailwindcss/typography` and retry.")
+  (spit "tailwind.config.cjs" (slurp (io/resource "stylesheets/tailwind.config.js")))
+  (spit "input.css" (slurp (io/resource "stylesheets/viewer.css")))
+  (fs/create-dirs "build")
+  (spit "build/viewer.js" (slurp (get @config/!resource->url "/js/viewer.js")))
+  (doseq [{:keys [file viewer]} docs]
+    (spit (let [path (fs/path "build" (str/replace file #"\.(cljc?|md)$" ".edn"))]
+            (fs/create-dirs (fs/parent path))
+            (str path))
+          (pr-str viewer)))
+  (sh "npx" "tailwindcss"
+      "--input" "input.css"
+      "--config" "tailwind.config.cjs"
+      "--output" (str (fs/path out-path "viewer.css"))
+      "--minify")
+  (swap! config/!resource->url assoc "/css/viewer.css" "viewer.css"))
```

Comment:
We should content address the css (like we do for the images) so the file name should be sha512.css